InterfacePython/TasmanianSG.in.py

```diff
@@ -644,6 +645,74 @@ def makeWaveletGrid(self, iDimension, iOutputs, iDepth, iOrder=1, liLevelLimits=
 
         self.pLibTSG.tsgMakeWaveletGrid(self.pGrid, iDimension, iOutputs, iDepth, iOrder, pLevelLimits)
 
+    def makeFourierGrid(self, iDimension, iOutputs, iDepth, sType, liAnisotropicWeights=[], liLevelLimits=[]):
+        '''
+        creates a new sparse grid using a Fourier rule
+        discards any existing grid held by this class
+
+        iDimension: int (positive)
+              the number of inputs
+
+        iOutputs: int (non-negative)
+              the number of outputs
+
+        iDepth: int (non-negative)
+                controls the density of the grid, i.e.,
+                the offset for the tensor selection, the meaning of
+                iDepth depends on sType
+                Example 1: sType == 'iptotal' will give a grid that
+                           interpolates exactly all polynomials of
+                           degree up to and including iDepth
+                Example 2: sType == 'qptotal' will give a grid that
+                           integrates exactly all polynomials of degree
+                           up to and including iDepth
+
+        sType: string identifying the tensor selection strategy
+              'level'     'curved'     'hyperbolic'     'tensor'
+              'iptotal'   'ipcurved'   'iphyperbolic'   'iptensor'
+              'qptotal'   'qpcurved'   'qphyperbolic'   'qptensor'
+
+        liAnisotropicWeights: list or numpy.ndarray of weights
+                              length must be iDimension or 2*iDimension
+                              the first iDimension weights
+                                                       must be positive
+                              see the manual for details
+
+        '''
+        if (iDimension <= 0):
+            raise TasmanianInputError("iDimension", "ERROR: dimension should be a positive integer")
```

Comment:
* the exceptions must be tested
* inside `testsTSG.in.py` look for function `def testBasicException()`. You want to essentially repeat the first group of tests that relates to `makeSequenceGrid()`, but make it for `def makeFourierGrid()`